sdk/eventhub/event-hubs/src/sender.ts

```diff
@@ -154,7 +161,40 @@ export class EventHubProducer {
     if (!Array.isArray(eventData) && !(eventData instanceof EventDataBatch)) {
       eventData = [eventData];
     }
-    return this._eventHubSender!.send(eventData, { ...this._senderOptions, ...options });
+
+    // link message span contexts
+    let spanContextsToLink: SpanContext[] = [];
+    if (Array.isArray(eventData)) {
+      for (let i = 0; i < eventData.length; i++) {
+        const event: EventData = {...eventData[i], properties: {...eventData[i].properties}};
+        const messageSpan = createMessageSpan(options.parentSpan);
+        // since these message spans are created from same context as the send span,
+        // these message spans don't need to be linked.
+        instrumentEventData(event, messageSpan);
+        messageSpan.end();
+      }
+    } else if (eventData instanceof EventDataBatch) {
+      spanContextsToLink = eventData._messageSpanContexts;
+    }
+
+
+    const sendSpan = this._createSendSpan(options.parentSpan);
+    for (const spanContext of spanContextsToLink) {
+      sendSpan.addLink(spanContext);
+    }
+
+    try {
+      const result = await this._eventHubSender!.send(eventData, { ...this._senderOptions, ...options });
+      sendSpan.setStatus({code: CanonicalCode.OK});
+      return result;
+    } catch (err) {
+      sendSpan.setStatus({
```

Comment:
I love this stuff, we need to be more diligent about doing it in the other libraries. You have inspired me